### ISSUE TITLE
Undocumented feature: Sparse Embedding trained with SignSGD

### ISSUE THREAD

**Issue by AlexiaJM** [sapientinc/HRM#73]
The paper does not mention the Sparse Embedding trained with SignSGD. The code says "Adam ≈ SignSGD if gradient is very sparse", but why would the embedding of the puzzle-id be sparse? And why not use Adam if its approximately equivalent to SignSGD when the embedding is sparse?  

Could you share a paper that document the benefit of SignSGD for embeddings or was this found to slightly improve performance from hyperparameter-tuning?

**Comment by AlexiaJM** [comment]
thanks
